pkg/operator/controllers/machine/machine_controller_test.go

```diff
@@ -0,0 +1,280 @@
+				ProviderSpec: machinev1beta1.ProviderSpec{
+					Value: &runtime.RawExtension{
+						Raw: []byte(`{
+"apiVersion": "azureproviderconfig.openshift.io/v1beta1",
+"kind": "AzureMachineProviderSpec",
+"osDisk": {
+"diskSizeGB": 512
+},
+"image": {
+"publisher": "azureopenshift",
+"offer": "aro4"
+},
+"vmSize": "Standard_D8s_v3"
+}`)},
+				},
+			},
+		}
+
+		master2 := &machinev1beta1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo-hx8z7-master-2",
+				Namespace: machineSetsNamespace,
+				Labels:    map[string]string{"machine.openshift.io/cluster-api-machine-role": "master"},
+			},
+			Spec: machinev1beta1.MachineSpec{
+				ProviderSpec: machinev1beta1.ProviderSpec{
+					Value: &runtime.RawExtension{
+						Raw: []byte(`{
+"apiVersion": "azureproviderconfig.openshift.io/v1beta1",
+"kind": "AzureMachineProviderSpec",
+"osDisk": {
+"diskSizeGB": 512
+},
+"image": {
+"publisher": "azureopenshift",
+"offer": "aro4"
+},
+"vmSize": "Standard_D8s_v3"
+}`)},
+				},
+			},
+		}
+
+		worker0 := &machinev1beta1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo-hx8z7-worker-0",
+				Namespace: machineSetsNamespace,
+				Labels:    map[string]string{"machine.openshift.io/cluster-api-machine-role": "worker"},
+			},
+			Spec: machinev1beta1.MachineSpec{
+				ProviderSpec: machinev1beta1.ProviderSpec{
+					Value: &runtime.RawExtension{
+						Raw: []byte(fmt.Sprintf(`{
+"apiVersion": "azureproviderconfig.openshift.io/v1beta1",
+"kind": "AzureMachineProviderSpec",
+"osDisk": {
+"diskSizeGB": %v
+},
+"image": {
+"publisher": "%v",
+"offer": "aro4"
+},
+"vmSize": "%v"
+}`, diskSize, imagePublisher, vmSize))},
+				},
+			},
+		}
+
+		worker1 := &machinev1beta1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo-hx8z7-worker-1",
+				Namespace: machineSetsNamespace,
+				Labels:    map[string]string{"machine.openshift.io/cluster-api-machine-role": "worker"},
+			},
+			Spec: machinev1beta1.MachineSpec{
+				ProviderSpec: machinev1beta1.ProviderSpec{
+					Value: &runtime.RawExtension{
+						Raw: []byte(`{
+"apiVersion": "azureproviderconfig.openshift.io/v1beta1",
+"kind": "AzureMachineProviderSpec",
+"osDisk": {
+"diskSizeGB": 128
+},
+"image": {
+"publisher": "azureopenshift",
+"offer": "aro4"
+},
+"vmSize": "Standard_D4s_v3"
+}`)},
+				},
+			},
+		}
+
+		worker2 := &machinev1beta1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "foo-hx8z7-worker-2",
+				Namespace: machineSetsNamespace,
+				Labels:    map[string]string{"machine.openshift.io/cluster-api-machine-role": "worker"},
+			},
+			Spec: machinev1beta1.MachineSpec{
+				ProviderSpec: machinev1beta1.ProviderSpec{
+					Value: &runtime.RawExtension{
+						Raw: []byte(`{
+"apiVersion": "azureproviderconfig.openshift.io/v1beta1",
+"kind": "AzureMachineProviderSpec",
+"osDisk": {
+"diskSizeGB": 128
+},
+"image": {
+"publisher": "azureopenshift",
+"offer": "aro4"
+},
+"vmSize": "Standard_D4s_v3"
+}`)},
+				},
+			},
+		}
```

Comment:
This differs from `worker1` only by name (`foo-hx8z7-worker-2`). To make this easier to read, can we do something like assign a `machinev1beta1.MachineSpec` variable above and use it twice, rather than repeating all this?

Same goes for the masters, of which all three are identical aside from the names. Let's just say it once.

Reply:
https://github.com/Azure/ARO-RP/issues/1605